tracer/src/Datadog.Trace/Tagging/AzureServiceBusTags.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="AzureServiceBusTags.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.SourceGenerators;
+
+#pragma warning disable SA1402 // File must contain single type
+namespace Datadog.Trace.Tagging
+{
+    // Re-implement InstrumentationTags so the built-in Activity can copy over SpanKind
+    // without being limited by our tags infrastructure
```

Comment:
Hmmm, that seems annoying 🤔 I wonder if there's an easy-ish way around that which doesn't require re-implementation.

I guess you could old school it?
```csharp
internal partial class AzureServiceBusTags : InstrumentationTags
{
    private string _spanKind;

    [Tag(Trace.Tags.SpanKind)]
    public string SpanKind => _spanKind

    public void SetSpanKind(string spanKind) => _spanKind = spanKind;
```

Is that preferable? 🤔 

EDIT: I guess not, because I can't see where we copy across the tags, so I guess it's automatic? 🤔

Reply:
Unfortunately, this won't work for us either 😞 The problem is that using our Activity pipeline we automatically add a `span.kind` tag that corresponds to the Activity.ActivityKind, and we only do this when we're closing the Activity/span, not at the creation: https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs#L38

We might consider updating some of our tags infrastructure for this as we add more OTEL instrumentation since the span.kind should always be set for these tags